### ISSUE TITLE
Add default `[profile.release]` values to Cargo.toml if not present

### ISSUE THREAD

**Issue by ascjones** [paritytech/cargo-contract#40]
As a follow up to #33, which enables amending the `Cargo.toml`, we could add the following section if it is not present:

```
[profile.release]
panic = "abort"
lto = true
opt-level = "z"
overflow-checks = true
```

These are the default optimal settings for contract binary size. However if any of these settings are already defined in the Cargo.toml, then those values should be used instead of these defaults. Potentially the user could be warned if using a non-optimal setting.

**Comment by Robbepop** [comment]
Note that
- `overflow-checks = true` add overhead to the final Wasm binary size but is something like safe-math in Ethereum. However, it is safe-math overtuned because it applies safe-math everywhere instead of locally in user provided code. So more overhead than needed which is sub optimal.
- `codegen-units = 1` could be useful since it greatly further optimized the final Wasm, however, takes a lot more time to actually compile.
- `lto = true` is similar to `codegen-units = 1` in that it further optimizes at the cost of compilation.
- `panic = abort` is always needed.
- `opt-level = "z"` has shown in the past to be inferior to `opt-level = 3` when it comes to resulting Wasm file size. Has to be benchmarked and tested.

So we might end up with 2 main configurations:
1. `debug` configuration that tries to build quickly
2. `release` configuration that tries to apply most optimizations

**Comment by ascjones** [comment]
Closed by #55
